Multiview Dataset MuPPI
=======================
In this multi-view dataset we describe the proteins of a protein-protein interaction (PPI) network. The aim of building this multi-view dataset is to classify the extremely multifunctional1 (EMF) proteins from other proteins, and to identify their possible characteristics. The PPI network and the list of EMFs are taken from the work of Charles Chapple and Christine Brun [1], and are available from the MoonDB [2] database. The PPI network is generated by the MoonGO [1] pipeline gathering data from several different databases and filtering at the level of interaction identification methods to keep only interactions that can be considered as binary (i.e., likely direct). Self-interactions are discarded. This network is composed of 13683 proteins, and 90386 interactions.

Labels:
-------
| "EMF_labels.txt" file.
| mono_clustered : 10129 examples
| multi_clustered : 3316 examples
| EMF : 238 examples

Proteins are classified into three categories: mono cluster, multi cluster and EMF. These classes are obtained through the MoonGO [1] pipeline on the PPI network mentioned above. This pipeline defines the protein clusters in the graph, based on their Biological Process annotations, from Gene Ontolgy. Thus, culsters group proteins with similar biological roles. Some proteins belong to only one cluster, they are called "mono_cluster". Others belong to several, they are called "multi_cluster". Among the multi-clusters, proteins that have a very rare pair of anntotation terms, in the Gene Ontolgy database, are classified as extremely multifunctional (EMF).

PPI network topology view:
--------------------------
| “PPInetwork_topology.txt” file, and “View0” in the hdf5 file. 
| (13679 examples, 10 features)

This view describes the topological properties of the PPI network for each node. It is obtained via the "node table" function on cytoscape. Explanations of the features : https://med.bioinf.mpi-inf.mpg.de/netanalyzer/help/2.7/index.html#closenessCent

- AverageShortestPathLength -float- : 
The length of the shortest path between two nodes n and m is L(n,m). The shortest path length distribution gives the number of node pairs (n,m) with L(n,m) = k for k = 1,2,…
The network diameter is the maximum length of shortest paths between two nodes. If a network is disconnected, its diameter is the maximum of all diameters of its connected components.
The network diameter and the shortest path length distribution may indicate small-world properties of the analyzed network.

- BetweennessCentrality -float- : 
The betweenness centrality Cb(n) of a node n is computed as follows:
Cb(n) = ∑s≠n≠t (σst (n) / σst),
where s and t are nodes in the network different from n, σst denotes the number of shortest paths from s to t, and σst (n) is the number of shortest paths from s to t that n lies on.
Betweenness centrality is computed only for networks that do not contain multiple edges. The betweenness value for each node n is normalized by dividing by the number of node pairs excluding n: (N-1)(N-2)/2, where N is the total number of nodes in the connected component that n belongs to. Thus, the betweenness centrality of each node is a number between 0 and 1. The betweenness centrality of a node reflects the amount of control that this node exerts over the interactions of other nodes in the network. This measure favors nodes that join communities (dense subnetworks), rather than nodes that lie inside a community.

- ClusteringCoefficient -float- : 
In undirected networks, the clustering coefficient Cn of a node n is defined as Cn = 2en/(kn(kn-1)), where kn is the number of neighbors of n and en is the number of connected pairs between all neighbors of n. The clustering coefficient is a ratio N / M, where N is the number of edges between the neighbors of n, and M is the maximum number of edges that could possibly exist between the neighbors of n. The clustering coefficient of a node is always a number between 0 and 1.

- ClosenessCentrality -float- : 
The closeness centrality Cc(n) of a node n is defined as the reciprocal of the average shortest path length and is computed as follows:
Cc(n) = 1 / avg( L(n,m) ),
where L(n,m) is the length of the shortest path between two nodes n and m. The closeness centrality of each node is a number between 0 and 1.
Closeness centrality is a measure of how fast information spreads from a given node to other reachable nodes in the network.

- Eccentricity -int- : 
The maximum non-infinite length of a shortest path between n and another node in the network. If n is an isolated node, the value of this attribute is zero.

- Stress -int- : 
This attribute counts the number of shortest paths passing through a node.

- Degree -int- : 
In undirected networks, the node degree of a node n is the number of edges linked to n. A self-loop of a node is counted like two edges for the node degree. The node degree distribution gives the number of nodes with degree k for k = 0, 1, …

- NeighborhoodConnectivity -float- : 
The connectivity of a node is the number of its neighbors. The neighborhood connectivity of a node n is defined as the average connectivity of all neighbors of n. The neighborhood connectivity distribution gives the average of the neighborhood connectivities of all nodes n with k neighbors for k = 0, 1, ...

- Radiality -float- : 
This attribute is a node centrality index computed by subtracting the average shortest path length of a node n from the diameter of the connected component plus 1. The radiality of each node is divided by the diameter of the connected component. Thus it is a number between 0 and 1.

- TopologicalCoefficient -float- : 
The topological coefficient Tn of a node n with kn neighbors is computed as follows:
Tn = avg ( J(n,m) ) / kn.
Here, J(n,m) is defined for all nodes m that share at least one neighbor with n. The value J(n,m) is the number of neighbors shared between the nodes n and m, plus one if there is a direct link between n and m. The topological coefficient is a relative measure for the extent to which a node shares neighbors with other nodes.

Subcell location view:
----------------------
| "Subcell_Location.txt" file, and "View1” in the hdf5 file.
| (10203 examples, 50 features)

| Cell localization data by Immunofluorescence for 10203 proteins, has been gathered from the Human Protein Atlas (HPA).
| Source: https://www.proteinatlas.org/about/download (file subcellular_location.tsv.zip)

- 50 Cell compartments -float-
The presence of a protein in a given cell compartment is described by five levels of confidence: Enhanced, Supported, Approved, Uncertain and None. These confidence levels are translated into probabilities ranging from 0 to 1.

Tissue Expression view:
-----------------------
| "Tissue_Expression.txt" file, and "View2” in the hdf5 file.
| (13351 examples, 50 features)

| This view collects RNAseq data for 13351 proteins in 50 tissues. It represents consensus RNAseq data of three datasets: HPA, Gtex and Fantom. For a protein in a given tissue, the expression level stored in the view is the highest among the three datasets.
| Source: https://www.proteinatlas.org/about/download  (file rna_tissue_consensus.tsv.zip)

- 50 Tissues -float-
Values are the normalized consensus expression levels.

PPInetwork Embedding view:
--------------------------
| "SAGE_PPInetwork.txt" file, and "View3” in the hdf5 file.
| (13683 examples, 128 features)

- This view was generated with a Graph Embedding algorithm (Graph SAGE [3]) with as input the PPI network and topology informations as node attributes, via the Pytorch Geometric library (https://pytorch-geometric.readthedocs.io/en/latest/). The obtained matrix is 13683x128, with the 13683 proteins described by a vector of 128 floats.

Gene Ontology view:
-------------------
| "Gene_Ontology_BP.txt" file, and "View4” in the hdf5 file.
| (12422 examples, 10817 features) ou version light : (12422, 6081)
| "Gene_Ontology_CC.txt" file, and "View5” in the hdf5 file.
| (13005 examples, 1743 features) ou version light : (13005, 896)
| "Gene_Ontology_MF.txt" file, and "View6” in the hdf5 file.
| (13220 examples, 3746 features) ou version light : (13220, 1433)

- This view describes 3 tables (one for each sub-graph GO: Cellular Component (CC), Biological Process (BP) and Molecular Function (MF)) with as index the PPI proteins, and as columns the GO Term of the sub-graph. For a given protein, a 1 indicates the protein is annotated for this GO Term or this GO Term is an ancestor of a protein annotated GO Term.. The other columns are filled with 0's. To define ancestors of a GO Term, the relationships taken into account are "is a", "part of" file, and "regulates". Perfectly identical columns are merged into one, as well as their column names. Thus three identical columns A, B, and C will be merged into one: A_B_C, containing the same information as the first three.
| The file that lists the GO annotations for each protein :
| ftp://ftp.ebi.ac.uk/pub/databases/GO/goa/UNIPROT/goa_uniprot_all.gaf.gz
| The Gene Ontology graph with all the relationships between the terms :
| http://purl.obolibrary.org/obo/go.obo available on: http://geneontology.org/docs/download-ontology/
| And the python library to browse the obo file and retrieve the parents of each : https://github.com/tanghaibao/goatools/blob/master/README.md

GO PPInetwork Embedding view:
-----------------------------
| "GO-BP_PPInetwork_embed.txt" file, and "View7” in the hdf5 file.
| (13683 examples, 128 features)
| "GO-CC_PPInetwork_embed.txt" file, and "View8” in the hdf5 file.
| (13683 examples, 128 features)

This view was generated with a Graph Embedding (Graph SAGE [3]) algorithm with the weighted PPI network as input, via the Pytorch Geometric library (torch-geometric.readthedocs.io/). The edge weighting is the Jaccard similarity coefficient between the Gene Ontology annotations of the two interacting proteins. Two embedding are performed, the results of two different weightings: one taking into account the Cellular Component annotations, the other the Biological Process annotations. The lists of annotations used for the calculation are those generated for the "Gene Ontology" view.

Phenotype Ontology view:
------------------------
| "Phenotype_Ontology.txt" file, and "View9” in the hdf5 file.
| (13683 examples, 9079 features) ou version light : (13683, 5673)

| The Human Phenotype Ontology (HPO) database provides a standardized vocabulary of phenotypic abnormalities encountered in human diseases. Each HPO term describes a phenotypic abnormality, such as blindness. The database is being developed using the medical literature, Orphanet, DECIPHER, and OMIM. The data are organized in a table with the PPI proteins as an index and the columns correspond to the terms HPO. Let's say that the gene coding for a protein A is annotated with a term X, and that this term X has the terms Y and Z as parents and ancestors. Then the table cells in row A and the X, Y and Z columns will be 1's.The table cells with no match will be 0's. The perfectly identical columns are merged into one, along with their column names. So three identical columns A, B and C will be merged into one: A_B_C, containing the same information as the first three. 4013 PPI proteins are encoded by genes annotated by HPO, the remaining 9670 have 0's in each term.
| Source annotations: https://hpo.jax.org/data/annotations (Genes to Phenotype)
| Source ontology: http://purl.obolibrary.org/obo (file hp.obo)

Protein Domains view:
---------------------
| "Protein_Domains.txt" file, and "View10” in the hdf5 file.
| (13428 examples, 12477 features) ou version light : (13439, 1646)

| The InterPro database characterizes and catalogs protein domains present in known proteins and references them with a unique identifier (ID). In this dataset the lines are the PPI proteins and the features are the Interpro IDs of the domains. This dataset is a count table, so each value corresponds to the number of domains in the column that the protein in the row has. Perfectly identical columns are merged into one, along with their column names. Thus three identical columns A, B and C will be merged into one: A_B_C, containing the same information as the first three.
| Source : http://www.ebi.ac.uk/interpro/download/
| (file protein2ipr.dat.gz for protein annotations)
| (file ParentChildTreeFile.txt for Interpro ontology's tree)

Post Translational Modifications view:
--------------------------------------
| "Post_Traductionnal_Modifications.txt" file, and "View11” in the hdf5 file.
| (13282 examples, 32 features)

| dbPTM data that catalogues post-translational modifications (PTM) on known proteins. In this dataset the rows are the PPI proteins and the columns are the different PTMs. The value in a column for a given protein is the number of PTM sites in that protein.
| Source : http://dbptm.mbc.nctu.edu.tw/download.php (All downloadable files from section “The statistics of experimental and putative PTM sites in dbPTM”)

3’UTR Complexes view:
---------------------
| "3UTR_Complexes.txt" file, and "View12” in the hdf5 file.
| (13683 examples, 11 features)

This view uses the data produced by the work of Ribeiro D. et al [5]. It consists of a list of proteins complexes : 3'UTR where an RNA recruits an RBP (RNA Binding Protein), which in turn recruits an intermediate protein, which in turn binds to the nascent protein being translated, from the starting RNA. These complexes are predicted from networks of protein-protein and RNA-protein interactions. The dataset obtained has the PPI proteins as rows and the three types of proteins in the complexes (nascent, intermediate and RBP) as columns.

- Nascent -int-
Number of times a protein is present in a 3'UTR complex as a nascent.

- Intermediate -int-
Number of times a protein is present in a 3'UTR complex as an intermediate.

- RBP -int-
Number of times a protein is present in a 3'UTR complex as a RBP.

- Inter_linked -int-
When a protein is nascent in one or more complexes, number of different intermediates to which it binds.

- RBP_linked -int-
When a protein is nascent in one or more complexes, number of different RBPs to which it binds.

- all -int-
Number of case where the protein is nascent and a RBP in complexe can bind all alternative 3’UTR of the RNA coding to the nascent.

- long -int-
Number of case where the protein is nascent and a RBP in complexe can bind only the longest alternative 3’UTR of the RNA coding to the nascent.

single -int-
Number of case where the protein is nascent, and the RNA coding to this protein have a single 3’UTR, no alternative ones, that can bind by a RBP in complexes

- short -int-
Number of case where the protein is nascent and a RBP in complexe can bind only the shortest alternative 3’UTR of the RNA coding to the nascent.

- mixed -int-
Number of case where the protein is nascent and a RBP in complexe can bind a mixed of long and short alternative 3’UTR of the RNA coding to the nascent.

- count -int-
Number of alternative 3’UTR of the RNA coding to the protein

Linear Motifs view:
-------------------
| "Linear_Motifs.txt”, and “View13” in the hdf5 file.
| (13 628 examples, 315 features)

| This view collects information on short linear motifs, also known as eukaryotic linear motifs (SLiMs or ELMs), detected in protein sequences as defined by the ELM database [4]. They are predicted by the SLIMProb tool, with a threshold on the motifs mean disorder at 0,4. We keep only ELM classes that have a probability of occurrence under 0,001 in the human proteome, and that been experimentally verified in at least one human protein. Doing so, we have identified putative 163 different ELMs on 9879 PPI proteins.
| Source of ELM classes and ELM instances (for experimentally proved ones, i.e., true positives) : http://elm.eu.org/downloads.html

- 163 features of ELM classes -int-
Count of ELMs predicted for a protein in example and the ELM classe in feature.

- 163 features of ELM classe density -float-
Count of ELMs predicted for a protein and a given ELM classe, divided by the protein length.

- Protein Length -int-
Number of amino acids of the protein.

- 2 features of Disorder score -float-
Each amino acid (AA) of a protein have a disorder score, between 0 and 1. It is the probability of the AA to be in a disordered protein region, as opposed to the ordered regions that are protein domains. In those features, we have the average disorder score of each protein. There is two way to calculate it so we have one column for each disorder score.


References
----------
[1] : Chapple, C. E. et al. Extreme multifunctional proteins identified from a human protein interaction network. Nat. Commun. 6, 7412 (2015).

[2] : Ribeiro, D. M., Briere, G., Bely, B., Spinelli, L. & Brun, C. MoonDB 2.0: an updated database of extreme multifunctional and moonlighting proteins. Nucleic Acids Res. 47, D398–D402 (2019).
MoonDB : http://moondb.hb.univ-amu.fr/

[3] : D. Wang, P. Cui, W. Zhu, Structural deep network embedding, in: Proceedings of the 22nd International Conference on Knowledge Discovery and Data Mining, ACM, 2016, pp. 1225–1234.

[4] : Kumar M1, Gibson TJ1 et al. ELM-the eukaryotic linear motif resource in 2020. Nucleic Acids Res. 48, D296-D306 (2020).

<<<<<<< HEAD
[5] : Diogo M. Ribeiro, Christine Brun et al. The role of 3’UTR-protein complexes in the regulation of protein multifunctionality and subcellular localization. bioRxiv 784702; doi: https://doi.org/10.1101/784702
=======
[5] : Diogo M. Ribeiro, Christine Brun et al. The role of 3’UTR-protein complexes in the regulation of protein multifunctionality and subcellular localization. bioRxiv 784702; doi: http://sci-hub.tw/10.1101/784702
>>>>>>> 165f297bf7c1b5968e50e320c738a58d87fac1b6


